### PR TITLE
Add Automation Calculator

### PR DESCRIPTION
**Pull request by ZitaNemeckova**
TODO:

- [x] Add Ansible Analytics to navbar
- [x] only `superuser` can see the navbar item
- [x] Add special unique EmptyStates
- [x] use API to decide ErrorStates correctly/ Clean empty states
- [x] update ErrorStates buttons to navigate to Settings - removed for now
- [x] Automation Calculator table
- [x] Automation Calculator chart https://www.npmjs.com/package/react-json-chart-builder/v/1.14.3 - there's still some dep problems
- [x] Automation Calculator summary
- [x] translations - done-ish
- [x] pagination
- [x] sorting
- [x] filtering

Only superadmin can access this page and sees Analytics -> Reports in nav bar.

How it looks right now:
Screenshot 2023-03-28 at 16 22 32

Depends on https://github.com/ansible/awx/pull/13652

Fixes https://issues.redhat.com/browse/AA-1536
Fixes https://issues.redhat.com/browse/AA-1537

### CODE REVIEW

**Review by vidyanambiar — COMMENTED** [review]
Seeing this error on https://localhost:3002/ui_next/reports
image

File: frontend/awx/analytics/Reports/AutomationCalculator.tsx [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+    }
+    setSearchParams(updatedSearchParams.toString());
+  };
+
+  const getOffset = (page: string, perPage: string | AttributeType): number => {
+    return page === '0' ? 0 : (parseInt(page) - 1) * parseInt(perPage.toString());
+  };
+
+  const { data, isLoading } = useSWR<ReportDataResponse, boolean, any>(
+    `/api/v2/analytics/roi_templates/?limit=${searchParams.get('limit') || '6'}&offset=${getOffset(
```
[the remark this answers is not in the document]
Comment (ZitaNemeckova) [reply frontend/awx/analytics/Reports/AutomationCalculator.tsx:169]: So it's a backend specific issue as auditor is not allowed to do POST requests. Martin will have a look on Tue.

**Comment by ZitaNemeckova** [comment]
@marshmalien Fixed! Given the deadline it was decided that auditor will not be allowed to see this page yet. They will after API is fixed in a follow-up PR.

File: frontend/awx/analytics/Reports/TemplatesTable/types.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  failed_hosts_costs: number;
+  manual_effort_minutes: number;
+  // Calculated fields
+  //delta: number;
+  //avgRunTime: number;
+  //manualCost: number;
+  //automatedCost: number;
+  //enabled: boolean;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  monetary_gain?: number | any;
```
Comment (vidyanambiar) [inline frontend/awx/analytics/Reports/TemplatesTable/types.ts:23]: What is the reason we're allowing type `any` here?

File: frontend/awx/analytics/components/Chart/hydrateSchema.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * the schema and return a plain function with the hydrated schema
+ * which can be used. This would optimize the code much more, since
+ * we would not neet to run JSON stringify and parse + replace
+ * mthods all the time. However the slowdown is pretty small,
+ * so I left it as is for now.
+ *
+ * @param schema The stringified version of the schema.
+ * @returns The hidrated schema with passed variables.
+ */
+const hydrateSchema = (schema: ChartSchemaElement[]) => (props: unknown) => {
```
Comment (vidyanambiar) [inline frontend/awx/analytics/components/Chart/hydrateSchema.ts:13]: why is the props type `unknown`?

File: frontend/awx/analytics/components/Chart/index.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+const applyHiddenFilter = (chartData: ChartData, chartSeriesHidden: string[] = []): ChartData => ({
+  ...chartData,
+  series: chartData.series.map((series: ChartDataSerie) => ({
+    ...series,
+    hidden:
+      (!!series.serie[0].id || !!series.serie[0].host_id) &&
+      !!chartSeriesHidden.includes(
+        // eslint-disable-next-line no-prototype-builtins
+        series.serie[0].hasOwnProperty('host_id').toString() ||
```
Comment (vidyanambiar) [inline frontend/awx/analytics/components/Chart/index.tsx:65]: We can get rid of the eslint-disable comment by changing this to `(Object.prototype.hasOwnProperty.call(series.serie[0], 'host_id').toString()`

**Review by vidyanambiar — COMMENTED** [review]
In the empty/error state I see this skeleton component and it doesn't go away. Is this intended?
Screenshot 2023-04-06 at 10 04 07 AM

**Comment by ZitaNemeckova** [comment]
@vidyanambiar All code comments fixes. Thanks for suggestions :) 

As for the skeleton toolbar it's on purpose from my point of view. It's populated from API so if it fails I have no data to populate it and it stays as loading state. But I can ask UX for correct pattern. I believe there's no pattern for this case in the codebase yet. I'm open to the suggestion :)

**Comment by h-kataria** [comment]
@vidyanambiar @mabashian is this PR ready to be merged, I believe one edge case situation is not resolved but it is not blowing up or anything. It just needs UX related issue.

**Pull request merged**